plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -55,13 +55,14 @@ def keycloak_argument_spec():
     :return: argument_spec dict
     """
     return dict(
-        auth_keycloak_url=dict(type='str', aliases=['url'], required=True, no_log=False),
+        auth_keycloak_url=dict(type='str', aliases=['url'], required=True),
         auth_client_id=dict(type='str', default='admin-cli'),
-        auth_realm=dict(type='str', required=True),
+        auth_realm=dict(type='str', default=None),
```

Comment:
```suggestion
        auth_realm=dict(type='str'),
```
Same for the other `default=None` you are adding.